internal/provider/resources/deployment_access.go

```diff
@@ -0,0 +1,349 @@
+package resources
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/listdefault"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/prefecthq/terraform-provider-prefect/internal/api"
+	"github.com/prefecthq/terraform-provider-prefect/internal/provider/customtypes"
+	"github.com/prefecthq/terraform-provider-prefect/internal/provider/helpers"
+)
+
+var _ = resource.ResourceWithConfigure(&DeploymentAccessResource{})
+
+type DeploymentAccessResource struct {
+	client api.PrefectClient
+}
+
+type DeploymentAccessResourceModel struct {
+	AccountID   customtypes.UUIDValue `tfsdk:"account_id"`
+	WorkspaceID customtypes.UUIDValue `tfsdk:"workspace_id"`
+
+	DeploymentID customtypes.UUIDValue `tfsdk:"deployment_id"`
+
+	ManageActorIDs types.List `tfsdk:"manage_actor_ids"`
+	RunActorIDs    types.List `tfsdk:"run_actor_ids"`
+	ViewActorIDs   types.List `tfsdk:"view_actor_ids"`
+	ManageTeamIDs  types.List `tfsdk:"manage_team_ids"`
+	RunTeamIDs     types.List `tfsdk:"run_team_ids"`
+	ViewTeamIDs    types.List `tfsdk:"view_team_ids"`
+}
+
+// NewDeploymentAccessResource returns a new DeploymentAccessResource.
+//
+//nolint:ireturn // required by Terraform API
+func NewDeploymentAccessResource() resource.Resource {
+	return &DeploymentAccessResource{}
+}
+
+// Metadata returns the resource type name.
+func (r *DeploymentAccessResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_deployment_access"
+}
+
+// Configure initializes runtime state for the resource.
+func (r *DeploymentAccessResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(api.PrefectClient)
+	if !ok {
+		resp.Diagnostics.Append(helpers.ConfigureTypeErrorDiagnostic("resource", req.ProviderData))
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *DeploymentAccessResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	defaultEmptyList, _ := basetypes.NewListValue(types.StringType, []attr.Value{})
+
+	resp.Schema = schema.Schema{
+		Description: "The resource `deployment_access` represents a connection between an accessor " +
```

Comment:
i love using the markdown ```` ticks, but it does prevent us from writing more sane multiline strings here in the go layer

Reply:
Yeah I'm with you there. Could define them elsewhere in the file so they're more "left-aligned" 🤔